ignition-server/controllers/local_ignitionprovider.go

```diff
@@ -451,3 +472,70 @@ func copyFile(src, dst string) error {
 	}
 	return os.Chmod(dst, srcinfo.Mode())
 }
+
+// findImageRefByArch finds the appropriate image reference in a multi-arch manifest image based on the current platform's OS and processor architecture
+func findImageRefByArch(ctx context.Context, imageRef string, pullSecret []byte, osToFind string, archToFind string) (manifestImageRef string, err error) {
+	manifestList, err := registryclient.GetManifest(ctx, imageRef, pullSecret)
+	if err != nil {
+		return "", fmt.Errorf("failed to retrieve manifest from image ref, %s: %w", imageRef, err)
+	}
+
+	_, payload, err := manifestList.Payload()
+	if err != nil {
+		return "", fmt.Errorf("failed to get manifest payload: %w", err)
+	}
+
+	m := new(manifestlist.DeserializedManifestList)
+	if err = m.UnmarshalJSON(payload); err != nil {
+		return "", fmt.Errorf("failed to get unmarshalled manifest list: %w", err)
+	}
+
+	matchingManifestForArch, err := findMatchingManifest(imageRef, m, osToFind, archToFind)
+	if err != nil {
+		return "", fmt.Errorf("failed to retrieve matching manifest for os/arch, %s/%s: %w", osToFind, archToFind, err)
+	}
+
+	return matchingManifestForArch, nil
+}
+
+// findMatchingManifest looks to find a manifest matching the current platform's OS and processor architecture from a deserialized manifest list from an image's payload
+func findMatchingManifest(imageRef string, deserializedManifestList *manifestlist.DeserializedManifestList, osToFind string, archToFind string) (string, error) {
+	var foundManifestDesc = manifestlist.ManifestDescriptor{}
+	for _, manifestDesc := range deserializedManifestList.ManifestList.Manifests {
+		if osToFind == manifestDesc.Platform.OS && archToFind == manifestDesc.Platform.Architecture {
+			foundManifestDesc = manifestDesc
+			break
+		}
+	}
+
+	if reflect.DeepEqual(foundManifestDesc, manifestlist.ManifestDescriptor{}) {
+		return "", fmt.Errorf("not found")
```

Comment:
You can do
```
var foundManifestDesc *manifestlist.ManifestDescriptor
	if foundManifestDesc == nil {
		return "", fmt.Errorf("not found")
	}
```

Reply:
Yeah that looks cleaner than what I did. It's been updated.